data/tabular/scaffold_split.py

```diff
@@ -0,0 +1,182 @@
+    "blood_brain_barrier_martins_et_al",
+    "serine_threonine_kinase_33_butkiewicz",
+    "ld50_zhu",
+    "herg_blockers",
+    "clearance_astrazeneca",
+    "half_life_obach",
+    "drug_induced_liver_injury",
+    "kcnq2_potassium_channel_butkiewicz",
+    "sarscov2_3clpro_diamond",
+    "volume_of_distribution_at_steady_state_lombardo_et_al",
+    "sr_hse_tox21",
+    "nr_er_tox21",
+    "cyp2c9_substrate_carbonmangels",
+    "nr_aromatase_tox21",
+    "cyp_p450_2d6_inhibition_veith_et_al",
+    "cyp_p450_1a2_inhibition_veith_et_al",
+    "cyp_p450_2c9_inhibition_veith_et_al",
+    "m1_muscarinic_receptor_antagonists_butkiewicz",
+    "nr_ar_tox21",
+    "sr_atad5_tox21",
+    "tyrosyl-dna_phosphodiesterase_butkiewicz",
+    "cav3_t-type_calcium_channels_butkiewicz",
+    "clintox" "sr_p53_tox21",
+    "nr_er_lbd_tox21" "pampa_ncats",
+    "sr_mmp_tox21",
+    "caco2_wang",
+    "sarscov2_vitro_touret",
+    "choline_transporter_butkiewicz",
+    "orexin1_receptor_butkiewicz",
+    "human_intestinal_absorption",
+    "nr_ahr_tox21",
+    "cyp3a4_substrate_carbonmangels",
+    "herg_karim_et_al",
+    "hiv",
+    "carcinogens",
+    "sr_are_tox21",
+    "nr_ppar_gamma_tox21",
+    "solubility_aqsoldb",
+    "m1_muscarinic_receptor_agonists_butkiewicz",
+    "ames_mutagenicity",
+    "potassium_ion_channel_kir2_1_butkiewicz",
+    "cyp_p450_3a4_inhibition_veith_et_al",
+    "skin_reaction",
+    "cyp2d6_substrate_carbonmangels",
+    "cyp_p450_2c19_inhibition_veith_et_al",
+    "nr_ar_lbd_tox21",
+    "p_glycoprotein_inhibition_broccatelli_et_al",
+    "bioavailability_ma_et_al",
+    "BACE",
+    "hiv",
+    "lipophilicity",
+    "thermosol",
+    "MUV_466",
+    "MUV_548",
+    "MUV_600",
+    "MUV_644",
+    "MUV_652",
+    "MUV_689",
+    "MUV_692",
+    "MUV_712",
+    "MUV_713",
+    "MUV_733",
+    "MUV_737",
+    "MUV_810",
+    "MUV_832",
+    "MUV_846",
+    "MUV_852",
+    "MUV_858",
+    "MUV_859",
+]
+
+
+def split_for_smiles(smiles: str, train_smiles: List[str], val_smiles: List[str]):
+    if smiles in train_smiles:
+        return "train"
+    elif smiles in val_smiles:
+        return "valid"
+    else:
+        return "test"
+
+
+def main(
+    data_dir,
+    override: bool = False,
+    train_frac: float = 0.7,
+    val_frac: float = 0.15,
+    test_frac: float = 0.15,
+    seed: int = 42,
+    debug: bool = False,
+):
+    all_yaml_files = glob(os.path.join(data_dir, "**", "*.yaml"), recursive=True)
+    if debug:
+        all_yaml_files = all_yaml_files[:5]
+    transformed_files = []
+    for file in all_yaml_files:
+        with open(file, "r") as f:
+            meta = yaml.safe_load(f)
+
+        if meta["name"] in to_scaffold_split:
+            # run transform.py script in this directory if `data_clean.csv` does not exist
+            # use this dir as cwd
+            if not os.path.exists(
+                os.path.join(os.path.dirname(file), "data_clean.csv")
+            ):
+                subprocess.run(
```

Comment:
When you are running "transform.py" you are not ensuring that the additional mol. reprs. are present. But we can keep it as it is, but just to mention that here.